cmd/cstorvolumeclaim/start.go

```diff
@@ -27,23 +31,27 @@ import (
 	clientset "github.com/openebs/maya/pkg/client/generated/clientset/versioned"
 	informers "github.com/openebs/maya/pkg/client/generated/informers/externalversions"
 	ndmclientset "github.com/openebs/maya/pkg/client/generated/openebs.io/ndm/v1alpha1/clientset/internalclientset"
-	"github.com/openebs/maya/pkg/signals"
+	leader "github.com/openebs/maya/pkg/kubernetes/leaderelection"
 	kubeinformers "k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
 var (
-	masterURL  string
-	kubeconfig string
+	masterURL              string
+	kubeconfig             string
+	leaderElectionLockName = "cvc-controller-leader"
+)
+
+// Command line flags
+var (
+	leaderElection          = flag.Bool("leader-election", true, "Enables leader election.")
```

Comment:
can you give reference code for this in k8s?

Reply:
All the csi based controllers like resize-controller, snaphot-controller and provisioners are using this,

Reply:
can you give the link to that?

Reply:
- https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/leaderelection/leader_election.go
- https://github.com/kubernetes-csi/external-attacher/blob/27b83ca98e10a48f9d13aed4f060aec418fae1a2/cmd/csi-attacher/main.go#L175